### PR TITLE
lxc list, two new usage columns

### PR DESCRIPTION
**Pull request by erifri**
I want a more detailed view of resource consumption. 
I've added memory in percent (consumed memory/ limits.memory) and CPU.Usage in seconds.

Example below:

```
lxc list -c nsm,limits.memory,MU

+------------------+---------+--------------+---------------+---------------+-----------+
|       NAME       |  STATE  | MEMORY USAGE | LIMITS MEMORY | MEMORY USAGE% | CPU USAGE |
+------------------+---------+--------------+---------------+---------------+-----------+
| c1               | RUNNING | 63.44MB      | 512MB         | 12.4%         | 51s       |
+------------------+---------+--------------+---------------+---------------+-----------+
| c2               | RUNNING | 113.46MB     | 1024MB        | 11.1%         | 101s      |
+------------------+---------+--------------+---------------+---------------+-----------+
| stopped          | STOPPED |              | 1024MB        |               |           |
+------------------+---------+--------------+---------------+---------------+-----------+
| no_memory_limits | RUNNING | 461.75MB     |               |               | 1057s     |
+------------------+---------+--------------+---------------+---------------+-----------+

```

### CODE REVIEW

File: lxc/list.go [hunk 1]
```diff
@@ -663,6 +665,31 @@ func (c *cmdList) memoryUsageColumnData(cInfo api.InstanceFull) string {
 	return ""
 }
 
+func (c *cmdList) memoryUsagePercentColumnData(cInfo api.InstanceFull) string {
+
+	if cInfo.IsActive() && cInfo.State != nil && cInfo.State.Memory.Usage > 0 {
+
+		if cInfo.ExpandedConfig["limits.memory"] != "" {
```
Comment (stgraber) [inline lxc/list.go:674]: Just worth keeping in mind that this won't work for anyone using a memory limit in percents.

File: lxc/list.go [hunk 2]
```diff
@@ -432,6 +432,8 @@ func (c *cmdList) parseColumns(clustered bool) ([]column, bool, error) {
 		'F': {i18n.G("BASE IMAGE"), c.baseImageFullColumnData, false, false},
 		'l': {i18n.G("LAST USED AT"), c.LastUsedColumnData, false, false},
 		'm': {i18n.G("MEMORY USAGE"), c.memoryUsageColumnData, true, false},
+		'%': {i18n.G("MEMORY USAGE%"), c.memoryUsagePercentColumnData, true, false},
```
Comment (stgraber) [inline lxc/list.go]: I'd probably stay away from special characters and use `M` instead here.

File: lxc/list.go [hunk 3]
```diff
@@ -432,6 +432,8 @@ func (c *cmdList) parseColumns(clustered bool) ([]column, bool, error) {
...
+		'U': {i18n.G("CPU USAGE"), c.cpuUsageSecondsColumnData, true, false},
```
Comment (stgraber) [inline lxc/list.go:438]: Should be moved down (we're trying to keep alphabetical order) and probably just renamed to `u`.

File: lxc/list.go [hunk 4]
```diff
@@ -663,6 +665,31 @@ func (c *cmdList) memoryUsageColumnData(cInfo api.InstanceFull) string {
 	return ""
 }
 
+func (c *cmdList) memoryUsagePercentColumnData(cInfo api.InstanceFull) string {
+
```
Comment (stgraber) [inline lxc/list.go:671]: Remove leading white line or CI will complain :)

File: lxc/list.go [hunk 5]
```diff
@@ -663,6 +665,31 @@ func (c *cmdList) memoryUsageColumnData(cInfo api.InstanceFull) string {
...
+	if cInfo.IsActive() && cInfo.State != nil && cInfo.State.Memory.Usage > 0 {
+
```
Comment (stgraber) [inline lxc/list.go:673]: Remove leading white line or CI will complain :)

**Comment by stgraber** [comment]
On top of the few comments I listed, I also seem to remember adding new columns to `lxc list` requiring a change to `list_test.go`, adding the new letters to `shorthand` at least.

File: lxc/list.go (hunk 1 above)
Comment (erifri) [reply lxc/list.go:672]: I've added a check and for now return empty string if limit is in percent.

Elegant solution would be to calculate how much memory percentage represents and use that in the division below. I'm just not sure if that information is available in this function.

**Comment by stgraber** [comment]
Going to be a bit of rebase and cleaning up on this and then should be ready for review and merge.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
Thanks!

**Pull request merged**